server/routes/api/conversations.js

```diff
@@ -69,6 +69,15 @@ router.get("/", async (req, res, next) => {
 
       // set properties for notification count and latest message preview
       convoJSON.latestMessageText = convoJSON.messages[convoJSON.messages.length-1].text;
+      
+      //calculate unreadCount for conversation
+      convoJSON.unreadCount = convoJSON.messages.filter(message => message.senderId === convoJSON.otherUser.id && message.read === false).length;
+      
+      // calculate if each message is last read or not
+      const otherUserReadMessages = convoJSON.messages.filter((message) => message.senderId !== convoJSON.otherUser.id && message.read === true);
+      const lastRead = otherUserReadMessages.length ? otherUserReadMessages.reduce((prev, current) => (prev.id > current.id) ? prev : current) : null;
+      convoJSON.messages.map(message => lastRead && message.id === lastRead.id ? message.isLastReadByOtherUser = true : message.isLastReadByOtherUser = false);
```

Comment:
Generally map shouldn't be used if you're not using the array that is returned: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#when_not_to_use_map